### PR TITLE
Add integration test harness using Ferrum

### PR DESCRIPTION
**Pull request by joelhawksley**
### What

This PR introduces a proof-of-concept integration harness using Ferrum, leveraging ViewComponent previews to provide the example routes.

### Why

In order to test javascript behaviors, we need to test our ViewComponents in integration, in a browser environment. 

See https://github.com/primer/view_components/pull/250 for more detail.

### CODE REVIEW

**Comment by manuelpuyol** [comment]
@joelhawksley Does storybook and `yarn build` still work without webpack in `demo/`?

File: demo/package.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
   "dependencies": {
     "@rails/actioncable": "^6.0.0",
     "@rails/ujs": "^6.0.0",
```
Comment (manuelpuyol) [inline demo/package.json:7]: since we are here... do we need these packages?

**Comment by joelhawksley** [comment]
> @joelhawksley Does storybook and yarn build still work without webpack in demo/?

Just checked locally, and both still work 👍🏻

File: test/application_system_test_case.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+require "test_helper"
+require "capybara/rails"
+require "capybara/minitest"
+require "capybara/cuprite"
+
+Ferrum::Browser.new(process_timeout: 5)
+
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+  driven_by :cuprite, using: :chrome, screen_size: [1400, 1400]
```
Comment (manuelpuyol) [inline test/application_system_test_case.rb:11]: any reasoning behind 1400x1400?

File: test/application_system_test_case.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  def with_preview(preview_name)
+    component_uri = self.class.name.split("Integration").first.underscore
+
+    visit("/rails/view_components/#{component_uri}/#{preview_name}")
```
Comment (manuelpuyol) [inline test/application_system_test_case.rb:16]: we also have the stories routes available here! wdyt of using them? 
e.g: `/rails/stories/primer/avatar_stack_component/avatar_stack`

File: test/application_system_test_case.rb (hunk 2 above)
Comment (joelhawksley) [reply test/application_system_test_case.rb:11]: Nope, this was default from examples we saw.

File: test/application_system_test_case.rb (hunk 3 above)
Comment (joelhawksley) [reply test/application_system_test_case.rb:16]: Interesting. I could go either way. I think we'll want to define specific previews for specific integration tests, but I could also see us reusing the Storybook infrastructure. Ultimately they are similar concepts that should probably share more logic than they do in this PR. 

Perhaps we can land what we have here and look into this further at a later time?

**Review by manuelpuyol — APPROVED** [review]
We can talk more about previews vs storybook later 🚢

**Pull request merged**